lib/JSON/Validator.pm

```diff
@@ -719,8 +719,12 @@ sub _guessed_right {
 }
 
 sub _is_date_time {
-  $_[0] =~ qr/^(\d+)-(\d+)-(\d+)T(\d+):(\d+):(\d+(?:\.\d+)?)(?:Z|([+-])(\d+):(\d+))?$/io;
+  $_[0] =~ qr/
+    ^(\d+)-(\d+)-(\d+)\D+(\d+):(\d+):(\d+(?:\.\d+)?)   # Date and time
```

Comment:
I implemented a more conservative option, although I also considered `.`. I didn't choose `.` because:
1) I'm unsure if the wording means that the separator can be omitted entirely (and not just replaced)
2) If allowing a digit as a separator there is in the spirit of the RFC itself